app/controllers/coupons_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CouponsController < ApplicationController
+
+  def index
+    @coupons = Coupon.all
+  end
+
+  def new
+
+  end
```

Comment:
Not something I'd expect you to know yet, but when a controller action is empty, this syntax is preferred. 
```suggestion
  def new; end
```